src/util/Observer.hxx

```diff
@@ -62,27 +66,37 @@ public:
    * @brief   Adds an observer to listen to Subject's events
    * @param   observer the observer to add
    */
-  void addObserver(ObserverSPtr<Data> observer) { m_Observers.emplace_back(std::move(observer)); }
+  void addObserver(ObserverWPtr<Data> observer) { m_Observers.emplace_front(std::move(observer)); }
+
+  /**
+   * @brief Remove all non valid observers.
+   */
+  virtual void prune()
+  {
+    m_Observers.remove_if([](ObserverWPtr<Data> wPtr) { return wPtr.expired(); });
+  }
 
 private:
-  std::vector<ObserverSPtr<Data>> m_Observers;
+  std::forward_list<ObserverWPtr<Data>> m_Observers;
```

Comment:
Why did you change to linked list? I think it's slower no?

Reply:
if it was because of remove_if, you should use std::remove_if instead for vector

Reply:
linked lists are bad for cache locality... I'm not sure that we ever want them

Reply:
The remove_erase idiom is linear time just like remove_if for linked list in ::prune, so I don't see an advantage